src/event/trigger.js

```diff
@@ -99,8 +99,7 @@ jQuery.extend( jQuery.event, {
 				special.bindType || type;
 
 			// jQuery handler
-			handle = ( dataPriv.get( cur, "events" ) || {} )[ event.type ] &&
-				dataPriv.get( cur, "handle" );
+			handle = dataPriv.get( cur, "handle" );
```

Comment:
What was the reason for the `events` check here when we only use `handle`?

Reply:
Just an optimization I suspect. If there are no events of that type for this element, we can avoid the call to `handle`. However, the call to get the data object isn't free so it probably doesn't save anything. In the 1.7.x era this was a direct lookup on an object so it might have been worth the difference. Today I don't see it being important.

Reply:
Ah, so this is an optimization that prevents calling `handle` for elements that have some events data but not for an event we're triggering right now? The `handle` function is basically some simple checks & then calling `jQuery.event.dispatch`. That function is not particularly light so I'll need to check if it's worth removing this optimization.

BTW, `Object.prototype` collisions are not a big issue here as the worst that could happen is code triggering a colliding event name does a bit more work than necessary. That's why it'd be hard to write a test that fails on the `master` version of this code.

Reply:
I think it is worth keeping a guard in place to ensure that `handle` is empty if the event type is not registered.

Reply:
I restored the previous check, just with `Object.create( null )` as a fallback.